import-scripts/fetch-ddp-and-import-cmo-access-data.sh

```diff
@@ -0,0 +1,262 @@
+# 4. create meta files for the 3 timeline data files and copy timeine info into the data repository
+# 5. fix case list files by overwriting stable_id and cancer_study_identifier
+# 6. check importer compatibility with the database schema version
+# 7. import the study to the msk-portal
+# 8. clear the persistence cache for the msk-portal
+# 9. send email notification of study import
+# 10. commit changes/updates into the data repository clone
+# 11. push changeset to the data repository
+
+MY_FLOCK_FILEPATH="/data/portal-cron/cron-lock/fetch-ddp-and-import-cmo-access-data.lock"
+(
+    echo $(date)
+
+    # check lock so that executions of this script not overlap
+    if ! flock --nonblock --exclusive $my_flock_fd ; then
+        echo "Failure : could not acquire lock for $MY_FLOCK_FILEPATH -- the fetch from yesterday is still running."
+        exit 1
+    fi
+
+    source $PORTAL_HOME/scripts/dmp-import-vars-functions.sh
+    source $PORTAL_HOME/scripts/clear-persistence-cache-shell-functions.sh
+
+    PIPELINES_EMAIL_LIST="cbioportal-pipelines@cbioportal.org"
+    CANCER_STUDY_IDENTIFIER="mixed_msk_cfdna_research_access"
+    IMPORT_SYMLINK_FILEPATH="/data/portal-cron/cbio-portal-data/cmo-access/mixed_msk_cfdna_research_access"
+    now=$(date "+%Y-%m-%d-%H-%M-%S")
+    IMPORTER_JAR_FILENAME="$PORTAL_HOME/lib/msk-cmo-importer.jar"
+    ENABLE_DEBUGGING=0
+    JAVA_IMPORTER_ARGS="$JAVA_PROXY_ARGS $JAVA_SSL_ARGS -Dspring.profiles.active=dbcp -Djava.io.tmpdir=$CMO_ACCESS_TMPDIR -ea -cp $IMPORTER_JAR_FILENAME org.mskcc.cbio.importer.Admin"
+    cmo_access_notification_file=$(mktemp $CMO_ACCESS_TMPDIR/cmo-access-portal-update-notification.$now.XXXXXX)
+    ONCOTREE_VERSION_TO_USE="oncotree_candidate_release"
+    cmo_access_dmp_pids_filepath=$CMO_ACCESS_TMPDIR/cmo_access_patient_list.txt
+    cmo_access_seq_date_filepath=$CMO_ACCESS_TMPDIR/cmo_access_seq_date.txt
+    cmo_access_ddp_output_dirpath=$CMO_ACCESS_TMPDIR/ddp_output
+
+    # -----------------------------------------------------------------------------------------------------------
+    # START CMO-ACCESS DATA FETCHING
+    echo $(date)
+
+    if [ -z $JAVA_BINARY ] | [ -z $GIT_BINARY ] | [ -z $PORTAL_HOME ] | [ -z $CMO_ACCESS_DATA_HOME ] ; then
+        message="could not run fetch-ddp-and-import-cmo-access-data.sh: automation-environment.sh script must be run in order to set needed environment variables (like CMO_ACCESS_DATA_HOME, ...)"
+        echo $message
+        echo -e "$message" | mail -s "fetch-ddp-and-import-cmo-access-data.sh failed to run." $PIPELINES_EMAIL_LIST
+        sendPreImportFailureMessageMskPipelineLogsSlack "$message"
+        exit 1
+    fi
+
+    # create tmpdir if it does not exist, and wipe contents
+    if [[ -d "$CMO_ACCESS_TMPDIR" ]] ; then
+        if [[ "$CMO_ACCESS_TMPDIR" != "/" ]] ; then
+            rm -rf "$CMO_ACCESS_TMPDIR"/*
+        fi
+    else
+        mkdir "$CMO_ACCESS_TMPDIR" 
```

Comment:
where's CMO_ACCESS_TMPDIR defined?

Reply:
good catch. I did not check in my changes to automation_environment.sh ... this is defined there. I will add this to the branch.